TopMovies/MoviesConnector/TopMoviesConnector.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  TopMoviesConnector.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 10.02.2021.
+//
+
+import ReSwift
+
+class TopMoviesConnector: BaseConnector<TopMoviesProps> {
+  
+  required init(updateProps: @escaping (TopMoviesProps) -> Void) {
+    super.init(updateProps: updateProps)
+  }
+  
+  override func newState(state: MainState) {
+    switch state.movieCategoriesState.categoriesList {
+    case let .completed(categoriesID):
+      let props =
+        TopMoviesProps(
+          movieCategories:
+            categoriesID
+            .map{ MovieCategoryProps(
+              categoryNameText:
+                state.movieCategoriesState.relational[$0]?.title ?? "Category Name",
+              movies:
+                state.movieCategoriesState.relational[$0]?.movies
+                .map{ MovieCollectionProps(
+                  movie: state.moviesState.relational[$0] ?? Movie()
+                ) } ?? [] )
```

Comment:
If id is removed from relational, it is possible that this model was deleted or accidentally not stored.
That's why you'd rather map categories in props verifying that each ID is contained in relational. If not - no props should be created.

**Task**:  Rewrite that code using `Array.compactMap` and `Optional.map` methods

Reply:
@RomanTysiachnik can i use init? to resolve that problem?

Reply:
Try just to rewrite that code unwrapping optionals using high order functions, without using guard of if statements.